File: samples/client/3_0_3_unit_test/java/src/main/java/org/openapijsonschematools/client/schemas/validation/PatternValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.exceptions.ValidationException;

import java.util.regex.Pattern;

public class PatternValidator implements KeywordValidator {
    public final Pattern pattern;

    public PatternValidator(Pattern pattern) {
        this.pattern = pattern;
    }

    @Override
    public Object getConstraint() {
        return pattern;
    }

    @Override
    public PathToSchemasMap validate(JsonSchema schema, Object arg, ValidationMetadata validationMetadata, Object extra) {
        if (!(arg instanceof String)) {
            return null;
        }
        if (!pattern.matcher((String) arg).find()) {
            throw new ValidationException("Invalid value "+arg+" did not find a match for pattern "+pattern);
        }
        return null;
    }
}

File: samples/client/3_0_3_unit_test/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/PatternValidator.kt
package org.openapijsonschematools.client.schemas.validation

import org.openapijsonschematools.client.exceptions.ValidationException

import java.util.regex.Pattern

class PatternValidator : KeywordValidator {
    @Throws(ValidationException::class)
    override fun validate(
        data: ValidationData
    ): PathToSchemasMap? {
        val pattern: Pattern = data.schema.pattern ?: return null
        if (data.arg !is String) {
            return null
        }
        if (!pattern.matcher(data.arg).find()) {
            throw ValidationException("Invalid value ${data.arg} did not find a match for pattern $pattern")
        }
        return null
    }
}